crates/wasmtime/src/engine.rs

```diff
@@ -1,9 +1,93 @@
-use crate::Config;
-use std::sync::Arc;
+use crate::{Caller, Config, Func, FuncType, IntoFunc, Store, Trap, Val};
+use anyhow::Result;
+use std::any::Any;
+use std::collections::HashMap;
+use std::sync::{Arc, RwLock};
 #[cfg(feature = "cache")]
 use wasmtime_cache::CacheConfig;
 use wasmtime_jit::Compiler;
-use wasmtime_runtime::debug_builtins;
+use wasmtime_runtime::{debug_builtins, InstanceHandle, VMTrampoline};
+
+// This type needs to be public for `IntoFunc`, but is considered an internal implementation detail.
+#[doc(hidden)]
+pub struct EngineFunc {
+    pub(crate) ty: FuncType,
```

Comment:
To avoid the `pub(crate)` here could this perhaps be defined in the `func.rs` module?

Reply:
These fields are also accessed from `Store::get_engine_func`, but I'm happy to make these private with accessor impls if you think it's warranted.

Reply:
Oh true, perhaps though there could be something like `engine_func.to_func(&store)`? (mostly just looking to localize this logic if we can)

Reply:
I'll see what I can do to localize the implementation to `EngineFunc`.